src/App.js

```diff
@@ -8,35 +10,28 @@ import Products from './components/Products';
 import ShoppingCart from './components/ShoppingCart';
 
 function App() {
-	const [products] = useState(data);
-	const [cart, setCart] = useState([]);
-
-	const addItem = item => {
-		setCart([...cart, item]);
-	};
-
-	return (
-		<div className="App">
-			<Navigation cart={cart} />
-
-			{/* Routes */}
-			<Route
-				exact
-				path="/"
-				render={() => (
-					<Products
-						products={products}
-						addItem={addItem}
-					/>
-				)}
-			/>
-
-			<Route
-				path="/cart"
-				render={() => <ShoppingCart cart={cart} />}
-			/>
-		</div>
-	);
+    const [products] = useState(data);
+
+    const [cart, setCart] = useState([]);
+
+    const addItem = item => {
+        setCart([...cart, item]);
+    };
+    return (
+        <ProductContext.Provider value={{products, addItem}}>
+            <CartContext.Provider value={cart}>
+                <div className="App">
+                    <Navigation cart={cart}/>
+
+                    {/* Routes */}
+                    <Route exact path="/" component={Products}/>
+
+                    <Route path="/cart" component={ShoppingCart}/>
+                </div>
+            </CartContext.Provider>
+        </ProductContext.Provider>
```

Comment:
Try to be more explicit with your providers, having `ProductContext.Provider` wrap the whole app when only the Products component is making use of it makes it's true purpose unclear.